### PR TITLE
consolidate scattered ex-distribution infra and tools docs

### PR DESCRIPTION
**Pull request by bobheadxi**
A lot of infra docs were formerly under Distribution docs, which have been "deprecated" but not actually moved to a new home, even though a lot of this infrastructure is quite critical, especially to CI. This PR creates `/product-engineering/engineering/infrastructure` and `/product-engineering/engineering/tools` so that these things that will likely never really change much and will be used by many teams can survive team splits and renames, and tries to consolidate docs here. We can assign owners in these pages (as I have for devx and Buildkite) in subsequent PRs.

- remove duplicated internal infrastructure page in Cloud DevOps and Distribution
- move customer env replicas stuff previously in Distribution internal infrastructure into Delivery
- migrate docs related to internal infrastructure and internal tooling out of Distribution
- merge "Environments" page, which only talks about GCP, into a combined `infrastructure/gcp` page along with the commitments stuff formerly in Distribution
- Move slackgenie up into tools
- Move zoom recordings guide into `/communication` - everyone uses Zoom! Surprisingly couldn't find a better home for this
- adds docs about currently available queues as requested in https://github.com/sourcegraph/deploy-sourcegraph-dot-com/pull/15085#pullrequestreview-824491838 (trying to find a home for this information prompted this PR)
   - why here and not docs.sourcegraph.com? because this pertains to more than just `sourcegraph/sourcegraph`, and explicitly is a matter related to internal infra

### CODE REVIEW

**Comment by bobheadxi** [comment]
Neither the DevX team nor the Delivery team are taggable for review in this repo 🥲

File: content/product-engineering/engineering/enablement/delivery/customers/replicas.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Customer environment replicas
+
+We maintain separate AWS accounts with Sourcegraph instances and other infrastructure that mimic various customers' environments for testing purposes. See "[Customer environment replicas and managed instances](https://my.1password.com/vaults/dnrhbauihkhjs5ag6vszsme45a/003/ctqvj7zcmdiujmfh2mxzffdlym)" on 1Password for the list.
+
+## Accessing sourcegraph internal infrastrcture
+
+To access this AWS account:
+
+1. Request an AWS user ID by asking a member of the #distributioneers team in slack.
```
Comment (caugustus-sourcegraph) [inline content/product-engineering/engineering/enablement/delivery/customers/replicas.md]:
```suggestion
1. Request an AWS user ID by asking in the #it-tech-ops channel in slack.
```

AWS access should be gated by Okta now so I think requests fall on IT.

**Review by caugustus-sourcegraph — APPROVED** [review]

**Review by dan-mckean — APPROVED** [review]
I've not reviewed all the changes themselves - but I'm totally on board with the premise of the changes as summarized.

**Comment by daxmc99** [comment]
giphy

Amazing work! Thanks for moving this over!

**Pull request merged**